pymongo/bulk.py

```diff
@@ -149,8 +149,19 @@ def _merge_command(
 
 def _raise_bulk_write_error(full_result: _DocumentOut) -> NoReturn:
     """Raise a BulkWriteError from the full bulk api result."""
+    # retryWrites on MMAPv1 should raise an actionable error.
     if full_result["writeErrors"]:
         full_result["writeErrors"].sort(key=lambda error: error["index"])
+        err = full_result["writeErrors"][0]
+        code = err["code"]
+        msg = err["errmsg"]
+        if code == 20 and msg.startswith("Transaction numbers"):
+            errmsg = (
+                "This MongoDB deployment does not support "
+                "retryable writes. Please add retryWrites=false "
+                "to your connection string."
+            )
+            raise OperationFailure(errmsg, code, full_result)
```

Comment:
We should raise BulkWriteError here instead of OperationFailure. Each writeError with "Transaction numbers" + code 20 we can replace the errmsg  be replaced with "This MongoDB deployment does support...".

Reply:
Actually I was incorrect. I was testing the behavior of mmapv1 but this new code handles a different case, when a shard is a standalone mongod. I added a test case here: https://github.com/mongodb/mongo-python-driver/pull/1606